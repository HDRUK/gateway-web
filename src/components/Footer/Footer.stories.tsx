import type { Meta, StoryObj } from "@storybook/react";
import Footer from "./Footer";

const meta: Meta<typeof Footer> = {
    component: Footer,
    title: "Layout/Footer",
    tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof Footer>;

export const Default: Story = {};
