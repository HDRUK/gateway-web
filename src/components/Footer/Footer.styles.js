import { css } from "@emotion/react";

export const footer = css({
    fontSize: "14px",
    backgroundColor: "#29235c",
});

export const footerContainer = css({
    color: "white",
    padding: "50px 20px",
});

export const list = css({
    listStyle: "none",
    textDecoration: "none",
    gap: "20px",
    paddingLeft: 0,
    "@media (min-width: 420px)": {
        display: "flex",
    },
});

export const social = css({
    display: "flex",
    alignItems: "center",
    gap: "6px",
});

export const link = css({
    textDecoration: "none",
    color: "white",
    "&:hover": {
        textDecoration: "underline",
    },
});

export const copyright = css({
    color: "#bdbada",
});
