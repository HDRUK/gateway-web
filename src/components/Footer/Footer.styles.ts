import { styled } from "@mui/material";
import Container from "@/components/Container";
import Link from "@/components/Link";

export const FooterWrapper = styled("footer")(({ theme }) => ({
    fontSize: "14px",
    ul: {
        listStyleType: "none",
        paddingLeft: 0,
    },
    background: "#fff",
}));

export const FooterContainer = styled(Container)(({ theme }) => ({
    color: "white",
    padding: "50px 20px",
    display: "flex",
    gap: "200px",
    [theme.breakpoints.down("tablet")]: {
        flexDirection: "column",
        gap: "16px",
    },
}));

export const FooterSocialLinks = styled("ul")(({ theme }) => ({
    textDecoration: "none",
    gap: "20px",
    paddingLeft: 0,
    display: "flex",
    [theme.breakpoints.down("tablet")]: {
        flexDirection: "column",
        gap: "16px",
    },
}));

export const FooterLinks = styled("ul")({
    display: "flex",
    flexDirection: "column",
    gap: "12px",
});

export const FooterLinksContainer = styled("ul")(({ theme }) => ({
    display: "flex",
    gap: "40px",
    [theme.breakpoints.down("tablet")]: {
        flexDirection: "column",
        gap: "16px",
    },
}));

export const FooterSocial = styled("span")({
    display: "flex",
    alignItems: "center",
    gap: "6px",
    color: "black",
});

export const FooterLink = styled(Link)({
    textDecoration: "none",
    color: "black",
    "&:hover": {
        textDecoration: "underline",
    },
});

export const FooterCopyright = styled("div")({
    color: "#000",
});
