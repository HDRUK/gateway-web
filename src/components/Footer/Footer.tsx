import { useTranslation } from "next-i18next";

const Footer = () => {
    const { t } = useTranslation("components");
    const copyright = String.fromCodePoint(0x00a9);
    const currentYear = new Date().getFullYear();
    return (
        <div style={{ textAlign: "center", width: "100%", padding: "20px" }}>
            {copyright}
            {t("Footer.text", { year: currentYear })}
        </div>
    );
};

export default Footer;
