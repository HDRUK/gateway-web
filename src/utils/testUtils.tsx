import React, { ReactNode } from "react";
import {
    render,
    renderHook,
    RenderHookOptions,
    RenderOptions,
    RenderResult,
    RenderHookResult,
} from "@testing-library/react";
import { SWRConfig } from "swr";
import { CacheProvider, ThemeProvider } from "@emotion/react";
import ActionBarProvider from "@/providers/ActionBarProvider";
import DialogProvider from "@/providers/DialogProvider";
import theme from "@/config/theme";
import createEmotionCache from "@/config/createEmotionCache";

const clientSideEmotionCache = createEmotionCache();

const Wrapper = ({ children }: { children: ReactNode }) => {
    return (
        <SWRConfig
            value={{
                provider: () => new Map(),
            }}>
            <CacheProvider value={clientSideEmotionCache}>
                <ThemeProvider theme={theme}>
                    <ActionBarProvider>
                        <DialogProvider>{children}</DialogProvider>
                    </ActionBarProvider>
                </ThemeProvider>
            </CacheProvider>
        </SWRConfig>
    );
};

interface OptionProps
    extends RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRender = (
    ui: React.ReactElement<
        unknown,
        string | React.JSXElementConstructor<unknown>
    >,
    options?: OptionProps
): RenderResult => {
    const { wrapperProps, ...rest } = options || {};

    return render(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

interface OptionHookProps
    extends RenderHookOptions<
        unknown,
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRenderHook = (
    ui: (initialProps: unknown) => unknown,
    options?: OptionHookProps
): RenderHookResult<unknown, unknown> => {
    const { wrapperProps, ...rest } = options || {};
    return renderHook(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

export * from "@testing-library/react";

export { customRenderHook as renderHook };
export { customRender as render };
