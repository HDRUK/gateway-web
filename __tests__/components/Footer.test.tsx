import React from "react";
import Footer from "@/components/Footer";
import { render } from "../testUtils";

describe("Footer", () => {
    it("should render component", async () => {
        const wrapper = render(<Footer />);

        expect(wrapper.container).toMatchSnapshot();
    });
});
