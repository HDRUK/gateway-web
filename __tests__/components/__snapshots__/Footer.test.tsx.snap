// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render component 1`] = `
<div>
  <div
    style="text-align: center; width: 100%; padding: 20px;"
  >
    Â©
    Footer.text
  </div>
</div>
`;
